server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1277,10 +1283,12 @@ private Engine.GetResult innerGet(Engine.Get get, boolean translogOnly, Function
         if (indexSettings.getIndexVersionCreated().isLegacyIndexVersion()) {
             throw new IllegalStateException("get operations not allowed on a legacy index");
         }
-        if (translogOnly) {
-            return getEngine().getFromTranslog(get, mappingLookup, mapperService.documentParser(), searcherWrapper);
-        }
-        return getEngine().get(get, mappingLookup, mapperService.documentParser(), searcherWrapper);
+        return withMutableEngine(engine -> {
```

Comment:
Real-time gets should work either with IndexEngine or HollowIndexEngine. We just need to ensure that the engine stays intact while the operations is ongoing (so there's no ACE thrown). So I think we'd need withMutableEngineOrNull here.